azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest.java

```diff
@@ -225,14 +234,8 @@ public void exec1FailedKillAll() throws Exception {
     Assert.assertTrue(runner.isKilled());
 
     Assert.assertTrue(
-        "Expected flow " + Status.FAILED + " instead " + exFlow.getStatus(),
-        exFlow.getStatus() == Status.FAILED);
-
-    try {
-        Thread.sleep(500);
```

Comment:
For that I added this commit:
Convert `FlowRunnerTest` to use custom job type instead of java process
- Using `InteractiveTestJob` instead of `azkaban.executor.SleepJavaJob`.
- This makes the test run a bit faster already.

Then I tried to continue from there, and got the test to run quite quickly by removing all the extra waiting time. But the test now reveals some rare race condition that I wasn't able to hunt down yet. The flow runner is getting stuck occasionally.